gtsfm/evaluation/compare_metrics.py

```diff
@@ -0,0 +1,80 @@
+"""Function supporting metric comparisons across SfM pipelines (GTSfM, COLMAP, etc.)
+
+This function converts outputs from SfM pipelines into a format matching
+metrics from GTSfM.
+
+Authors: Jon Womack
+"""
+import os
+from typing import Dict, List
+
+import gtsfm.utils.io as io_utils
+from gtsfm.evaluation.metrics import GtsfmMetric, GtsfmMetricsGroup
+
+import thirdparty.colmap.scripts.python.read_write_model as colmap_io
+
+
+
+def compute_metrics_from_txt(cameras, images, points3d):
+    """Calculate metrics from pipeline outputs parsed from COLMAP txt format.
+    Args:
+        cameras: dictionary of COLMAP-formatted Cameras
+        images: dictionary of COLMAP-formatted Images
+        points3D: dictionary of COLMAP-formatted Point3Ds
+    Returns:
+        other_pipeline_metrics: A dictionary of metrics from another pipeline that are comparable with GTSfM
+    """
+    cameras, images, image_files, sfmtracks = io_utils.colmap2gtsfm(cameras, images, points3d, load_sfmtracks=True)
+    num_cameras = len(cameras)
+    track_lengths = []
+    image_id_num_measurements = {}
+    for track in sfmtracks:
+        track_lengths.append(track.number_measurements())
+        for k in range(track.number_measurements()):
+            image_id, uv_measured = track.measurement(k)
+            if image_id not in image_id_num_measurements:
+                image_id_num_measurements[image_id] = 1
+            else:
+                image_id_num_measurements[image_id] += 1
+
+    other_pipeline_metrics = {
+        "number_cameras": GtsfmMetric("number_cameras", num_cameras),
+        "3d_tracks_length": GtsfmMetric(
+            "3d_tracks_length",
+            track_lengths,
+            plot_type=GtsfmMetric.PlotType.HISTOGRAM,
+        ),
+    }
+    return other_pipeline_metrics
+
+def save_other_pipelines_metrics(
+    txt_metric_paths: Dict[str, str],
+    json_path: str,
+    metric_filenames: List[str],
```

Comment:
nit: alternative (hopefully more clear) variable names: 
colmap_format_outputs (these are not metrics)
gtsfm_json_path
gtsfm_metrics_filenames

Reply:
This json_path contains jsons from gtsfm and other pipelines metrics, but I've modified the other variable names.